src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2121,8 +2123,30 @@ ManageWorkerPool(WorkerPool *workerPool)
 		/* create a session for the connection */
 		session = FindOrCreateWorkerSession(workerPool, connection);
 
+		pollMode = PQconnectPoll(connection->pgConn);
```

Comment:
What if we call ConnectionStateMachine here?

Reply:
yes. this is right.  call ConnectionStateMachine(session) is well.

Reply:
i have rewrite the code used ConnectionStateMachine.